Added touch swipe to side bar

ayush015 opened the pull request.

LalitNM commented on views/footer_imports.ejs:
```diff
@@ -1,4 +1,8 @@
+<script src="https://code.jquery.com/jquery-3.2.1.slim.min.js" integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN" crossorigin="anonymous"></script>
 <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js" integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q" crossorigin="anonymous"></script>
-<script src="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/js/bootstrap.min.js" integrity="sha384-JjSmVgyd0p3pXB1rRibZUAYoIIy6OrQ6VrjIEaFf/nJGzIxFDsf4x0xIM+B07jRM" crossorigin="anonymous"></script>
+<script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/js/bootstrap.min.js" integrity="sha384-JZR6Spejh4U02d8jOt6vLEHfe/JQGiRRSQQxSfFWpi1MquVdAyjUar5+76PVCmYl" crossorigin="anonymous"></script>
+<script src='https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.3/jquery.min.js'></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.touchswipe/1.6.19/jquery.touchSwipe.min.js" integrity="sha512-YYiD5ZhmJ0GCdJvx6Xe6HzHqHvMpJEPomXwPbsgcpMFPW+mQEeVBU6l9n+2Y+naq+CLbujk91vHyN18q6/RSYw==" crossorigin="anonymous"></script>
+<script src='/notify.min.js'></script>
```
Check this code again and add only those cdns, which are necessary. Remove all redundant cdns including `notify.min.js` as this is deleted in latest version of this repository.

LalitNM commented on views/head_imports.ejs:
```diff
@@ -4,4 +4,4 @@
 <script src="https://cdnjs.cloudflare.com/ajax/libs/socket.io/1.4.5/socket.io.js"></script>
-<script src="/client.js"></script>
\ No newline at end of file
+<script src="/client.js"></script>
```
I don't see any change in this file, please do not make unnecessary changes.

LalitNM commented on views/index.ejs:
```diff
@@ -44,8 +44,8 @@
-      </div>
-      <div class="right active" id="lobby-msg" data-joined="1">
+      </div> 
+      <div class="swipe-area right active" id="lobby-msg" data-joined="1">
```
adding `swipe-area` class here adds swipe action in lobby-room, but this action is not working in any room other than lobby. You need to add this in `client.js` too.

ayush015 replied:
I have made the required changes .

LalitNM reviewed (changes requested):
In addition to these changes, now the modal doesn't collapses after hitting `Add Room` button. Rectify this also.

Do these changes and then I will again review your pull request.

LalitNM committed:
Styled message input area with bootstrap

Changes Done
-Added code of improved textarea in index.ejs, client.js
-removed unnecessary styling from style.css
-modified function to send msg in script.js

LalitNM commented on public/script.js:
```diff
@@ -178,3 +178,15 @@ function notify(data, type){
+$(".swipe-area").swipe({
+  swipeStatus:function(event, phase, direction, distance, duration, fingers)
+    {
+      if (phase=="move" && direction =="right") {
+         $(".left").addClass("open-menu");
+      }
+      if (phase=="move" && direction =="left") {
+         $(".left").removeClass("open-menu");
+      }
```
In addition to these, you have to change class of `#toggle-icon` also. You have to change its class from `fa-angle-right` to `fa-angle-left` and vice-versa.

ayush015 replied:
Okay Yeah I'll do that .

LalitNM commented on public/client.js:
```diff
@@ -80,7 +80,6 @@ const appendUserInfo = (room_name, description) => {
     $('.card-columns').append($userInfo);
 
 }
-
```
remove this unnecessary change.

ayush015 replied:
That's just a line added without any code, anyways i'll revert it !  😀

LalitNM reviewed (changes requested):
And few more changes you have to do

- remove `package-lock.json` from commit.

- squash all commits in a single commit to make cleaner commit history.

ayush015 commented:
All the required changes are made !

LalitNM commented:
Great! Now only squash left.

ayush015 commented:
> Great! Now only squash left.

I'm getting unnecessary issues while squashing it .Can we just merge it ?  😅

LalitNM commented:
I'm sure that there is a way to clean this commit history, can you come on gitter I can explain you there. Just drop a message in 'Git' room on gitter. Follow [this](https://gitter.im/Baat-Cheet-IIT) link.

ayush015 commented:
> I'm sure that there is a way to clean this commit history, can you come on gitter I can explain you there. Just drop a message in 'Git' room on gitter. Follow [this](https://gitter.im/Baat-Cheet-IIT) link.

Done !

ayush015 committed:
squashed commits

ayush015 committed:
fixed textarea

ayush015 committed:
fixed non-clickable textarea bug

LalitNM commented on views/index.ejs:
```diff
@@ -45,7 +45,7 @@
         </div>
       </div>
-      <div class="swipe-area right active" id="lobby-msg" data-joined="1">
+      <div class="right active" id="lobby-msg" data-joined="1">
```
No need to remove this swipe-area.

LalitNM commented on views/index.ejs:
```diff
@@ -13,7 +13,7 @@
   <%- include('user', {title: title}); %>
   <div class="wrapper">
-    <div class="app-container">
+    <div class="app-container swipe-area">
```
instead remove this one.

ayush015 replied:
then it wont work with other rooms only lobby would work

LalitNM replied:
For other room you are adding this class in client.js.

ayush015 committed:
fixed non-clickable textarea bug

ayush015 committed:
fixed non-clickable textarea bug

LalitNM commented:
@ayush015 the problem is still there, but don't worry I will rectify that. I'm a bit busy nowadays.

ayush015 commented:
> @ayush015 the problem is still there, but don't worry I will rectify that. I'm a bit busy nowadays.

Okay its totally cool .So should I close this PR?

ayush015 committed:
fixed non-clickable textarea bug

LalitNM commented:
No, leave it , let me approve your changes.

ayush015 commented:
> No, leave it , let me approve your changes.

I have resolved the merge conflict Should i push it again?
In the meanwhile i'll also look for the solution !

LalitNM commented:
I resolved them for now. You can search for some solution and if you find it, then you should push it.

ayush015 commented:
> I resolved them for now. You can search for some solution and if you find it, then you should push it.

Yeah totally cool .I'll try find something and then i'll push it

ayush015 commented:
> > I resolved them for now. You can search for some solution and if you find it, then you should push it.
> 
> Yeah totally cool .I'll try find something and then i'll push it

ayush015 commented:
@LalitNM Hey man I have found a way we can resolve this . Just need little of your help !
Just want to be clear on the issue! I'm writing the issue below just let me know if this is the issue or its something else

So while we start the app the text area is not usable we cannot click and are not able to type anything right ?

LalitNM commented:
@ayush015 Yes, we cannot focus it by clicking on it. Also try to push the send button, this will focus the textarea, because the function behind that button focuses the textarea.

ayush015 commented:
> @ayush015 Yes, we cannot focus it by clicking on it. Also try to push the send button, this will focus the textarea, because the function behind that button focuses the textarea.

I have found a different approach . Its working !

ayush015 committed:
non clickable area bug

LalitNM commented:
@ayush015, I was just wondering that why you added a cdn just for one swipe, I don't think we will be using another function from that library. I want something similar to [this](https://stackoverflow.com/questions/15084675/how-to-implement-swipe-gestures-for-mobile-devices). And please rebase your branch so that you don't have to resolve conflicts, or you can make another pull request.

LalitNM committed:
Removed unnecessary changes

LalitNM committed:
removed jquery cdn

LalitNM committed:
removed unnecessary changes

LalitNM committed:
removed unnecessary changes

LalitNM committed:
removed unnecessary changes

Also added menu-closed class to leftbar

LalitNM committed:
Added better way of using touch gestures

ayush015 commented:
> @ayush015, I was just wondering that why you added a cdn just for one swipe, I don't think we will be using another function from that library. I want something similar to [this](https://stackoverflow.com/questions/15084675/how-to-implement-swipe-gestures-for-mobile-devices). And please rebase your branch so that you don't have to resolve conflicts, or you can make another pull request.

hey @LalitNM i'm busy with some other work right now . I'll get to it later if that's fine with you !

LalitNM committed:
Merge branch 'master' into ayushsri

LalitNM commented:
> hey @LalitNM i'm busy with some other work right now . I'll get to it later if that's fine with you !

No need to worry about this @ayush015, I did required changes and merged this pull request. You can look at some other issue.
By the way, it was great what you did :clap:
